src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal.hidden === false && animal instanceof Animal) {
```

Comment:
you can remove health only if animal instanceof `Herbivore` but not any Animal

Reply:
you forgot about this condition
<img width="829" alt="image" src="https://github.com/mate-academy/js_herbivores_and_carnivores/assets/6568745/3b5375a1-15d1-495c-b607-6506e2a12e43">